src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -852,12 +847,9 @@ private LLVMValueRef LLVMFunctionForMethod(MethodDesc callee, StackEntry thisPoi
 
         private LLVMValueRef GetOrCreateMethodSlot(MethodDesc method)
         {
-            var globalRefName = "__getslot__" + _compilation.NameMangler.GetMangledMethodName(method);
-            LLVMValueRef slot = LLVM.GetNamedGlobal(Module, globalRefName);
-            if(slot.Pointer == IntPtr.Zero)
-            {
-                slot = LLVM.AddGlobal(Module, LLVM.Int32Type(), globalRefName);
-            }
+            var vtableSlotSymbol = ((WebAssemblyCodegenNodeFactory)_compilation.NodeFactory).VTableSlot(method);
```

Comment:
It might be a general improvement to add `public new WebAssemblyCodegenNodeFactory NodeFactory` property to `WebAssemblyCodegenCompilation` so that this cast is not necessary. `WebAssemblyCodegenCompilationBuilder` can just pass it as a strongly typed argument to the constructor and Compilation can save it.